tests/interfaces/test_batch_autograd_qnode.py

```diff
@@ -0,0 +1,1238 @@
+# Copyright 2018-2020 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Integration tests for using the autograd interface with a QNode"""
+import pytest
+from pennylane import numpy as np
+
+import pennylane as qml
+from pennylane.beta import qnode, QNode
+from pennylane.tape import JacobianTape
+
+qubit_device_and_diff_method = [
+    ["default.qubit", "finite-diff", "backward"],
+    ["default.qubit", "parameter-shift", "backward"],
+    ["default.qubit", "backprop", "forward"],
+    ["default.qubit", "adjoint", "forward"],
+    ["default.qubit", "adjoint", "backward"],
+]
+
+
+@pytest.mark.parametrize("dev_name,diff_method,mode", qubit_device_and_diff_method)
+class TestQNode:
+    """Test that using the QNode with Autograd integrates with the PennyLane stack"""
+
+    def test_nondiff_param_unwrapping(self, dev_name, diff_method, mode, mocker):
+        """Test that non-differentiable parameters are correctly unwrapped
+        to NumPy ndarrays or floats (if 0-dimensional)"""
+        if diff_method != "parameter-shift":
+            pytest.skip("Test only supports parameter-shift")
+
+        dev = qml.device("default.qubit", wires=1)

```

Comment:
Will `dev_name` eventually be something other than `default.qubit`? I notice in the parameter array it is always the same, and furthermore `default.qubit` is being set manually set here.

Reply:
Maybe! Now that lightning.qubit is part of PL, we could potentially add lightning qubit to our tests